libobs/obs-output.c

```diff
@@ -188,8 +189,12 @@ void obs_output_destroy(obs_output_t *output)
 
 		if (output->service)
 			output->service->output = NULL;
-		if (output->context.data)
-			output->info.destroy(output->context.data);
+
+		if (output->context.data) {
+			void* temp_data = output->context.data;
+			output->context.data = NULL;
+			output->info.destroy(temp_data);
```

Comment:
Confused on which issue this fixing and why

Reply:
It for crashes on pthread_mutex_unlock( it called from pthread_mutex_destroy ). 
Usually it is a symptom of access to an object(which have a mutex as a member) what already being deleted/destroyed. 
In this change I clean pointer to a data before destroying it. So some other function could not use it while destroy is in process.

Reply:
I see, sounds good. I suspect there are still some other race conditions but this looks like it could fix some cases 👍